Code/GraphMol/FindStereo.cpp

```diff
@@ -0,0 +1,420 @@
+//
+//  Copyright (C) 2020 Greg Landrum and Rational Discovery LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <GraphMol/RDKitBase.h>
+#include <RDGeneral/Ranking.h>
+#include <GraphMol/new_canon.h>
+#include <RDGeneral/types.h>
+#include <algorithm>
+#include <RDGeneral/utils.h>
+#include <RDGeneral/Invariant.h>
+#include <RDGeneral/RDLog.h>
+
+#include <boost/dynamic_bitset.hpp>
+#include <boost/format.hpp>
+#include "Chirality.h"
+
+namespace RDKit {
+namespace Chirality {
+#ifndef _MSC_VER  
+const unsigned StereoInfo::NOATOM = std::numeric_limits<unsigned>::max();
+#endif
+namespace detail {
+
+StereoInfo getStereoInfo(const Bond *bond) {
+  PRECONDITION(bond, "bond is null");
+  StereoInfo sinfo;
+  const auto beginAtom = bond->getBeginAtom();
+  const auto endAtom = bond->getEndAtom();
+  if (bond->getBondType() == Bond::BondType::DOUBLE) {
+    if (beginAtom->getDegree() < 2 || endAtom->getDegree() < 2 ||
+        beginAtom->getDegree() > 3 || endAtom->getDegree() > 3) {
+      throw ValueErrorException("invalid atom degree in getStereoInfo(bond)");
```

Comment:
Might consider using boost::format to give a bit more info, like atom or bond index.

Reply:
I thought about that, but it's not really information you'll be able to do that much with. This test really could almost be a PRECONDITION since this is an internal-use-only function